knock/src/lib/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react';
+
+const useCopyToClipboard = (): [boolean, (text: string) => void] => {
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+
+  const copyToClipboard = (text: string) => {
+    navigator.clipboard.writeText(text).then(
+      () => {
+        setIsCopied(true);
+        setTimeout(() => setIsCopied(false), 2000);
+      },
+      () => {
+        setIsCopied(false);
+      },
+    );
+  };
```

Comment:
promise가 reject와 resolve를 활용할 때 다음과 같이 사용하는게 좀 더 가독성이 좋더라구요!
```suggestion
  const copyToClipboard = (text: string) => {
    navigator.clipboard.writeText(text).then(
      () => {
        setIsCopied(true);
        setTimeout(() => setIsCopied(false), 2000);
      }).catch(() => {
        setIsCopied(false);
      });
```